studio/components/interfaces/Database/Indexes/CreateIndexSidePanel.tsx

```diff
@@ -0,0 +1,240 @@
+import Link from 'next/link'
+import { useEffect, useMemo, useState } from 'react'
+import { Button, Input, Listbox, SidePanel } from 'ui'
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import CodeEditor from 'components/ui/CodeEditor'
+import MultiSelect, { MultiSelectOption } from 'components/ui/MultiSelect'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { useSchemasQuery } from 'data/database/schemas-query'
+import { useTableColumnsQuery } from 'data/database/table-columns-query'
+import { useEntityTypesQuery } from 'data/entity-types/entity-types-infinite-query'
+import { INDEX_TYPES } from './Indexes.constants'
+import { useExecuteSqlMutation } from 'data/sql/execute-sql-mutation'
+import { useIndexesQuery } from 'data/database/indexes-query'
+import { useStore } from 'hooks'
+
+interface CreateIndexSidePanelProps {
+  visible: boolean
+  onClose: () => void
+}
+
+const CreateIndexSidePanel = ({ visible, onClose }: CreateIndexSidePanelProps) => {
+  const { ui } = useStore()
+  const { project } = useProjectContext()
+  const [selectedSchema, setSelectedSchema] = useState('public')
+  const [selectedEntity, setSelectedEntity] = useState('---')
+  const [selectedColumns, setSelectedColumns] = useState<string[]>([])
+  const [selectedIndexType, setSelectedIndexType] = useState<string>(INDEX_TYPES[0].value)
+
+  const { refetch: refetchIndexes } = useIndexesQuery({
+    schema: selectedSchema,
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const { data: schemas } = useSchemasQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const { data: entities } = useEntityTypesQuery(
+    {
+      schema: selectedSchema,
+      sort: 'alphabetical',
+      search: undefined,
+      projectRef: project?.ref,
+      connectionString: project?.connectionString,
+    },
+    { keepPreviousData: true }
+  )
+  const {
+    data: tableColumns,
+    isLoading: isLoadingTableColumns,
+    isSuccess: isSuccessTableColumns,
+  } = useTableColumnsQuery({
+    table: 'countries',
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+
+  const { mutate: execute, isLoading: isExecuting } = useExecuteSqlMutation({
+    onSuccess() {
+      refetchIndexes()
```

Comment:
Can we `await refetchIndexes()` so the sidebar doesn't close before the UI has finished updating?